### PR TITLE
Add support for Marking and Unmarking tasks

### PR DESCRIPTION
**Pull request by DanzaSeah**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/task/AssignTaskCommand.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        Person personToEdit = lastShownPersonList.get(personIndex.getZeroBased());
+
+        boolean isVendor = false;
+        for (Vendor vendor : model.getFilteredVendorList()) {
+            if (personToEdit.getName().equals(vendor.getName())) {
+                isVendor = true;
+                break;
+            }
+        }
+        if (!isVendor) {
```
Comment (HanB1n) [inline src/main/java/seedu/address/logic/commands/task/AssignTaskCommand.java:90]: Good checking to ensure task assignment can only be done to Vendors.

File: src/main/java/seedu/address/logic/commands/task/AssignTaskCommand.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            if (taskIndex.getZeroBased() >= lastShownTaskList.size()) {
+                throw new CommandException(MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+            }
+            Task newTask = lastShownTaskList.get(taskIndex.getZeroBased());
+            if (updatedTasks.contains(newTask)) {
+                throw new CommandException(String.format(MESSAGE_DUPLICATE_TASK,
+                        newTask.toString(), personToEdit.getName().toString()));
+            }
+            tasksToAdd.add(newTask);
+            updatedTasks.add(newTask);
```
Comment (HanB1n) [inline src/main/java/seedu/address/logic/commands/task/AssignTaskCommand.java:106]: If you simply add the new task to the updatedTasks, it does not change the existing task. Would this result in duplicated tasks for a person?

File: src/main/java/seedu/address/logic/commands/task/TaskStatusModifier.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                        + targetIndex.getOneBased());
+            }
+            Task taskToModify = lastShownList.get(targetIndex.getZeroBased());
+            if (markAsDone) {
+                taskToModify.markAsDone();
+            } else {
+                taskToModify.markAsUndone();
+            }
+            modifiedTasks.add(taskToModify);
+            updatePersonsWithTask(model, personList, taskToModify);
```
Comment (HanB1n) [inline src/main/java/seedu/address/logic/commands/task/TaskStatusModifier.java:68]: LGTM

**Review by HanB1n — COMMENTED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/task/AssignTaskCommand.java (hunk 2 above)
Comment (DanzaSeah) [reply src/main/java/seedu/address/logic/commands/task/AssignTaskCommand.java:106]: Yeap, there's a check before this for duplicate tasks and an exception will be thrown before the task can be added

**Review by dasha3412 — APPROVED** [review]
LGTM, some fixes that we can add in separate PR

**Pull request merged**